aws-lc-rs/src/agreement.rs

```diff
@@ -513,7 +555,7 @@ impl<B: AsRef<[u8]>> UnparsedPublicKey<B> {
 #[inline]
 #[allow(clippy::needless_pass_by_value)]
 #[allow(clippy::missing_panics_doc)]
-pub fn agree_ephemeral<B: AsRef<[u8]>, F, R, E>(
+pub fn agree_ephemeral<B: AsRef<[u8]>, F, R, E: Copy>(
```

Comment:
`E: Copy` here is a breaking change to the API... I think it shouldn't be needed though if you refactor the map_err calls to `if let Err(()) = ... { return Err(error_value); }` or so -- the compiler can't track the ownership through the map_err but will be able to with if let.

Reply:
Good catch.  Will fix.